temporian/implementation/numpy/operators/set_index.py

```diff
@@ -76,28 +75,26 @@ def _append_impl(event: NumpyEvent, append_feat_names: List[str]) -> NumpyEvent:
     }
     # initialize destination event & sampling data
     dst_event_data = {}
-    for src_idx_lvl, src_idx_lvl_data in event.iterindex():
-        dst_idx_suffs = [
-            tuple(x)
-            for x in zip(*[src_idx_lvl_data.features[i] for i in dst_idx_pos])
-        ]
-        for target_dst_idx_suff in set(dst_idx_suffs):
-            # find all occurrences of destination suff in source event
-            dst_idx_suff_pos = [
-                idx
-                for idx, dst_idx_suff in enumerate(dst_idx_suffs)
-                if dst_idx_suff == target_dst_idx_suff
-            ]
+    for src_idx_key, src_idx_data in event.iterindex():
+        # constructing the dict of unique tuples and their positions
+        dst_idx_suffs = defaultdict(list)
+        for i, tpl in enumerate(
+            zip(*[src_idx_data.features[j] for j in dst_idx_pos])
+        ):
+            tpl = tuple(tpl)
+            dst_idx_suffs[tpl].append(i)
```

Comment:
How efficient is this? Raw for-loop in python is slow. Maybe there is an equivalent NP solution?

Reply:
Can't think of any right now. On the tests I ran, computing the destination suffixes and their positions beforehand was faster than traversing the features element by element (since you avoid multiple calls to `append` and instead do a single call to `IndexData` to create the destination features for each destination index key).